src/js/pensions/components/AdditionalSourcesField.jsx

```diff
@@ -0,0 +1,234 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import _ from 'lodash/fp';
+import Scroll from 'react-scroll';
+import { scrollToFirstError } from '../../common/utils/helpers';
+import { setItemTouched } from '../../common/schemaform/helpers';
+
+import {
+  toIdSchema,
+  deepEquals
+} from 'react-jsonschema-form/lib/utils';
+
+import { errorSchemaIsValid } from '../../common/schemaform/validation';
+
+const Element = Scroll.Element;
+const scroller = Scroll.scroller;
+
+function ReviewSources({ sources }) {
+  return (
+    <div>
+      {sources.map((source, index) =>
+        <div key={index} className="review-row"><dt>{source.name}</dt><dd>{source.amount}</dd></div>)}
+    </div>
+  );
+}
+
+export default class AdditionalSourcesField extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      editing: (props.formData || []).map(item => !item.name)
+    };
+
+    this.onItemChange = this.onItemChange.bind(this);
+    this.handleAdd = this.handleAdd.bind(this);
+    this.handleUpdate = this.handleUpdate.bind(this);
+    this.handleRemove = this.handleRemove.bind(this);
+    this.scrollToRow = this.scrollToRow.bind(this);
+  }
+
+  componentWillReceiveProps(newProps) {
+    const items = newProps.formData || [];
+    if (items.length < this.state.editing.length) {
+      this.setState({ editing: this.state.editing.filter((item, index) => index < items.length) });
+    } else if (items.length > this.state.editing.length) {
+      this.setState({ editing: this.state.editing.concat(true) });
```

Comment:
:man_shrugging: Doesn't really look like it, but how annoying would it be to mistype something, save it, realize you mistyped and...well bugger. Can't do anything about it now. Can't update it, can't delete it and start over. Meh.